src/hotspot/share/compiler/compilationPolicy.cpp

```diff
@@ -188,12 +188,12 @@ void CompilationPolicyUtils::Queue<InstanceKlass>::print_on(outputStream* st) {
   }
 }
 
-void CompilationPolicy::replay_training_at_init_loop(JavaThread* THREAD) {
-  while (!CompileBroker::is_compilation_disabled_forever()) {
-    InstanceKlass* ik = _training_replay_queue.pop(TrainingReplayQueue_lock, THREAD);
+void CompilationPolicy::replay_training_at_init_loop(JavaThread* current) {
+  while (!CompileBroker::is_compilation_disabled_forever() || AOTVerifyTrainingData) {
```

Comment:
Will it loop forever with `+ AOTVerifyTrainingData` ?

Reply:
Yes, it runs in a dedicated thread. It doesn't need to terminate.

Reply:
Add comment about this.

Reply:
I removed this with the last update, so it's not necessary.